packages allowlist can be defined from requirements like files

nlaurance-pyie opened the pull request:
because working with pip-compile generates requirements files with inline comments like
```
appdirs==1.4.4            # via black
black==20.8b1             # via -r requirements.in
certifi==2020.11.8        # via requests
```

and for now it's convenient to be able to paste it

nlaurance-pyie commented:
Packages and releases might be given as requirements.txt files

if requirements_path is missing it is assumed to be root folder

```ini
[plugins]
enabled =
    allowlist_requirements
[allowlist]
requirements_path = /my_folder
requirements =
    requirements.txt
```

nlaurance-pyie commented:
addresses issues#736

cooperlees commented on docs/filtering_configuration.md:
```diff
@@ -87,6 +87,21 @@ tag:tag:path.to.object =
+### requirements files Filtering
+Packages and releases might be given as requirements.txt files
+
+if requirements_path is missing it is assumed to be root folder
```
Root folder as in / or root folder configured in bandersnatch.conf

cooperlees commented on src/bandersnatch_filter_plugins/allowlist_name.py:
```diff
@@ -176,3 +192,46 @@ def _check_match(self, name: str, version_string: str) -> bool:
+        """
+        Parse the configuration file for
+        [allowlist]
+        requirements_path = /where_they_are
```
Ahh I see here.

cooperlees commented on src/bandersnatch_filter_plugins/allowlist_name.py:
```diff
@@ -176,3 +192,46 @@ def _check_match(self, name: str, version_string: str) -> bool:
+            requirement_line = requirement_line.strip()
+            if not requirement_line or requirement_line.startswith("#"):
+                continue
+            requirement_line, *_ = requirement_line.split('#')
```
```suggestion
            requirement_line, _ = requirement_line.split('#', maxsplits=1)
```
Big fan of max splits

nlaurance-pyie replied:
will do

cooperlees reviewed (changes requested):
Thanks! Nice testing. Please fix typing + black formatting too!

nlaurance-pyie commented:
It doesn't really work yet, that is we need both package and release filtering

nlaurance-pyie commented:
`pack, _ = 'package  # comment'.split('#', maxsplit=1)
pack, _ = 'package '.split('#', maxsplit=1)
Traceback (most recent call last):
  File "<input>", line 1, in <module>
ValueError: not enough values to unpack (expected 2, got 1)
`

cooperlees commented:
I normally just check if we need a split. but happy either way here.

`tox` will run all the lints + type checking locally. You have typing errors. We have pretty strict mypy settings.

cooperlees reviewed (approved):
Thanks for the contribution!
- Like to docker cleanups.
- Tests look pretty comprehensive to me

I’ll aim to release to PyPI this week. Maybe tonight. See how I roll. But I think the Docker Hub image will do for you.

This seems pretty specific - What’s your usage plan here? I’m just interested.
